### ISSUE TITLE
Cost updates: New cost defaulting behavior

### ISSUE THREAD

**Issue by kantai** [stacks-network/stacks-core#2874]
_Note:_ As discussed in [More Accurate Cost Functions for Clarity Native Function](https://forum.stacks.org/t/more-accurate-cost-functions-for-clarity-native-functions/12386) and recent blockchain meetings, one path forward for updating the costs is a Pre-2.1 fork that _only_ covers cost and block limit updates. Separate from the process of formally proposing, voting on, and confirming such a proposal is the task of actually implementing those changes. This issue is related only to the task of implementing those changes in the `next-costs` branch of this repository.

Currently, there is a single default "costs" contract: `SP000000000000000000002Q6VF78.costs` that is used as the default when the Clarity VM loads up the cost functions.

This "defaulting" behavior happens in [src/vm/costs/mod.rs#L639](https://github.com/blockstack/stacks-blockchain/blob/master/src/vm/costs/mod.rs#L639):

```rust
let boot_costs_id = boot_code_id("costs", self.mainnet);

...

for f in ClarityCostFunction::ALL.iter() {
    let cost_function_ref = cost_function_references.remove(&f).unwrap_or_else(|| {
        ClarityCostFunctionReference::new(boot_costs_id.clone(), f.get_name())
		});

...
```

Basically, if a cost vote hasn't defined a new cost function for a particular function, then a default is used (`boot_costs_id` refers to `SP0000..costs`).

In `next-costs`, this behavior needs to change. After the new epoch begins, if `SP000.costs-2-05` has been instantiated in the current fork, then the default should be *that* contract. If the new contract has not been instantiated, then the old default should be used.
